packages/views.api.unstable/src/parameters.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { AsyncValue } from "./utils/asyncValue.js";
+
+/**
+ * Map of the name of the type to the corresponding JavaScript type.
+ */
+type PrimitiveParameterTypes = {
+  boolean: boolean;
+  /** Formatted as an ISO date string, e.g. YYYY-MM-DD */
+  date: string;
+  number: number;
+  string: string;
+  /** Formatted as an ISO datetime string */
+  timestamp: string;
+};
+export type PrimitiveParameterType = keyof PrimitiveParameterTypes;
+
+export interface AbstractParameterValue<T extends PrimitiveParameterType> {
+  type: T;
+  value: AsyncValue<PrimitiveParameterTypes[T]>;
+}
+
+export type StringParameterValue = AbstractParameterValue<"string">;
+export type NumericParameterValue = AbstractParameterValue<"number">;
+export type BooleanParameterValue = AbstractParameterValue<"boolean">;
+export type DateParameterValue = AbstractParameterValue<"date">;
+export type TimestampParameterValue = AbstractParameterValue<"timestamp">;
+export type PrimitiveParameterValue =
+  | StringParameterValue
+  | NumericParameterValue
+  | BooleanParameterValue
+  | DateParameterValue
+  | TimestampParameterValue;
+
+export interface GenericArrayParameterValue<T extends PrimitiveParameterType> {
+  type: "array";
+  subType: T;
+  value: AsyncValue<PrimitiveParameterTypes[T][]>;
+}
```

Comment:
would it be cleaner to just have a `multiple: true` or similar on the AbstractParameterValue?

Reply:
I think you probably want an `optional: boolean` on the abstract type too?

Reply:
i think `multiple: true` is less legible in my head as to that meaning "array". for optional vs required, im still not sure which way we want to go (depends on how the host frame looks)